doc/source/data/iterating-over-data.rst

```diff
@@ -0,0 +1,273 @@
+.. _iterating-over-data:
+
+===================
+Iterating over data
+===================
+
+.. _iterating-over-rows:
+
+Iterating over rows
+===================
+
+To iterate over rows, call :meth:`Dataset.iter_rows() <ray.data.Dataset.iter_rows>`. Ray
+Data represents rows as dictionaries.
+
+.. testcode::
+
+    import ray
+
+    ds = ray.data.read_csv("s3://anonymous@air-example-data/iris.csv")
+
+    for row in ds.iter_rows():
+        print(row)
+
+.. testoutput::
+
+    {'sepal length (cm)': 5.1, 'sepal width (cm)': 3.5, 'petal length (cm)': 1.4, 'petal width (cm)': 0.2, 'target': 0}
+    {'sepal length (cm)': 4.9, 'sepal width (cm)': 3.0, 'petal length (cm)': 1.4, 'petal width (cm)': 0.2, 'target': 0}
+    ...
+    {'sepal length (cm)': 5.9, 'sepal width (cm)': 3.0, 'petal length (cm)': 5.1, 'petal width (cm)': 1.8, 'target': 2}
+
+
+For more information on working with rows, see
+:ref:`Transforming rows <transforming-rows>` and
+:ref:`Inspecting rows <inspecting-rows>`.
+
+.. _iterating-over-batches:
+
+Iterating over batches
+======================
+
+A batch contains data from multiple rows. To iterate over batches, call one of the
```

Comment:
Iterate over batches of dataset in different formats by calling one of the following methods: